smithy-aws-apigateway-openapi/src/main/java/software/amazon/smithy/aws/apigateway/openapi/AddIntegrations.java

```diff
@@ -121,6 +122,20 @@ private static ObjectNode getIntegrationAsObject(
         return integrationNode;
     }
 
+    private static void validateTraitConfiguration(IntegrationTrait trait,
+                                                   Context<? extends Trait> context,
+                                                   OperationShape operation) {
+        // For HTTP APIs, API Gateway requires that the payloadFormatVersion is set on integrations.
+        // https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-swagger-extensions-integration.html
+        // If the payloadFormatVersion has not been set on an integration and the apiGatewayType has been set to "HTTP",
+        // the conversion fails.
+        if (!trait.getPayloadFormatVersion().isPresent() && context.getConfig().getExtensions(ApiGatewayConfig.class)
```

Comment:
Nit: This if statement is a bit unwieldy, consider grabbing the `ApiType` before it and just doing the `.equals` comparison inside.

Reply:
Simplified this in `1084971`.